simnode/generator.py

```diff
@@ -67,9 +68,10 @@ def _interpolate_nodes(old_x_vals: List[float], old_y_vals: List[float], old_wid
             main_nodes = list(zip(new_x_vals, new_y_vals, z_vals, new_width_vals))
             main_road.nodes.extend(main_nodes)
             scenario.add_road(main_road)
-            # FIXME Recognize changing widths
+            # FIXME Recognize changing widths --- mostly done
             road_width = unique_nodes[0].width
             if road.markings:
+                # could theoretically be removed or integrated in _calculate_parallel_coords_list()
```

Comment:
This may be the case, but the method is already very long, so it should be a little more structured by giving parts of the commands names i.e. local functions.